src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ClosureConversion.cs

```diff
@@ -1597,17 +1598,33 @@ private BoundNode RewriteLambdaConversion(BoundLambda node)
                         // different from the local variable `type`, which has the node's type substituted for the current container.
                         var cacheVariableType = containerAsFrame.TypeMap.SubstituteType(node.Type).Type;
```

Comment:
>containerAsFrame.TypeMap.SubstituteType(node.Type).Type; [](start = 48, length = 56)

It looks like this duplicates the work that has already been done. It looks like this what we store in the ```type``` local.